scripts/xcode_cloud_helper.rb

```diff
@@ -1,762 +1,402 @@
-    XCode version used to create a new workflow
-
---issuer-id [issuer_id]:
-    Apple Connect API Issuer ID.
-
---key-id [key_id]:
-    Apple Connect API Key ID.
-
---pk-path [pk_path]:
-    Apple Connect API path to private key file.
-
---list-workflows:
-    Returns a list of current workflows for the RealmSwift product.
-
---list-products:
-    Returns a list of products associated to the Apple Connect Store account.
-
---list-repositories:
-    Returns a list of repositories integrated with XCode Cloud.
-
---list-mac-versions:
-    Returns a list of available mac versions.
-
---list-xcode-versions:
-    Returns a list of available xcode version.
-
---info-workflow [workflow_id]:
-    Returns the infor the corresponding workflow.
-
---create-workflow [name]:
-    Use with --xcode-version
-    Create a new workflow with the corresponding name.
-
---update-workflow [workflow_id]:
-    Updates workflow with the corresponding id.
-
---delete-workflow [workflow_id]:
-    Delete the workflow with the corrresponding id.
-
---build-workflow [workflow_id]:
-    Run a build for the corresponding workflow.
-
---create-new-workflows:
-    Adds the missing workflows corresponding to the list of targets and xcode versions in `pr-ci-matrix.rb`.
-
---clear-unused-workflows:
-    Clear all unused workflows which are not in the list of targets and xcode versions in `pr-ci-matrix.rb`.
-
---get-token:
-    Get Apple Connect Store API Token for local use.
-
-            EOF
-            exit
-        when '--token'
-            if arg == ''
-                raise "Token is required to execute this"
-            else
-                JWT_BEARER = arg
-            end
-        when '--team-id'
-            if arg != ''
-                TEAM_ID = arg
-            end
-        when '--issuer-id'
-            if arg != ''
-                issuer_id = arg
-            end
-        when '--key-id'
-            if arg != ''
-                key_id = arg
-            end
-        when '--pk-path'
-            if arg != ''
-                pk_path = arg
-            end
-        when '--info-workflow'
-            if arg != ''
-                workflow_id = arg
-            end
-        when '--create-workflow'
-            if arg != ''
-                name = arg
-            end
-        when '--delete-workflow', '--info-workflow', '--build-workflow', '--update-workflow'
-            if arg != ''
-                workflow_id = arg
-            end
-        when '--xcode-version'
-            if arg != ''
-                xcode_version = arg
-            end
-    end
+    $workflows_list ||= get_workflows
+    $workflows_list.find { |w| w['attributes']['name'].split('_')[0] == name }['id']
 end
 
-if JWT_BEARER == '' && option != '--get-token'
-    raise 'Token is needed to run this.'
-end
+Options = Struct.new(:token, :team_id, :issuer_id, :key_id, :pk_path)
+options = Options.new()
+$parser = OptionParser.new do |opts|
+    args = Options.new("world")
```

Comment:
I think this may not be necessary, we are creating a new Option Struct two lines above which is gonna be used to store the options arguments data, or maybe I'm missing something

Reply:
I think you're right.